### PR TITLE
chore: add ava config and some basic tests

### PR DESCRIPTION
**Pull request by yusefnapora**
This needs some cleanup, but I got ava + typescript to work after a bit of wrestling. 

So far I've only added some simple tests to keyring-core that don't depend on the service, but I've also started on a mock access service so we can test `registerSpace`, etc.

I had to add `"type": "module"` to the keyring-core package.json to get the ava + typescript setup to work. That seems pretty safe, since we're only including the rollup bundle in the published package, so it shouldn't make a difference to users.

It seems like the mock services would be useful for all the packages, so maybe we should make a shared `test-utils` package?

### CODE REVIEW

File: packages/keyring-core/test/utils/mock-service.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      return undefined
+    }),
+    recover: provide(Space.recover, async () => {
+      console.log('space/recover')
+      return []
+    })
+  }
+}
+
+export async function mockAcccessConnection (): Promise<ConnectionView<Service>> {
```
Comment (travis) [inline packages/keyring-core/test/utils/mock-service.ts]: extra "c" here

File: packages/keyring-core/test/agent.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import test from 'ava'
+import { JSDOM } from 'jsdom'
+import 'fake-indexeddb/auto'
+
+import { createAgent } from '../src/index.js'
```
Comment (travis) [inline packages/keyring-core/test/agent.ts]: we can get rid of these extensions if we use `[--experimental-specifier-resolution=node](https://nodejs.org/api/esm.html#customizing-esm-specifier-resolution-algorithm)` option mentioned in https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md#for-packages-with-type-module - I saw this earlier this week poking around at AVA but didn't get as far as you have here and am delighted to see it works locally. I'll push a commit doing this, happy to back it out if we don't want to use this

**Review by travis — APPROVED** [review]
this is fantastic!! I left a couple comments but already pushed fixes for them - happy to back either of my changes out if you don't want them, but I was in the code testing the `--experimental-specifier-resolution` change and it was easy to knock them both out

**Comment by travis** [comment]
this is already strictly better than what we have now so I'd be happy if you merged as-is!

> It seems like the mock services would be useful for all the packages, so maybe we should make a shared test-utils package?

I think it might be a little cleaner for each package to export useful test utils and to introduce a `services` package of some sort per https://github.com/web3-storage/w3ui/issues/156 where we can implement service Providers and land these mock services - I'm worried that a shared `test-utils` package will end up depending on all these packages, and that all the packages will end up dev-depending on `test-utils` leading to a gnarly knot of dependency resolution. Strictly speaking I think either will work, but the idea of a dependency knot like this puts a knot in my stomach...

**Comment by yusefnapora** [comment]
awesome, thanks for the fixes @travis :)

I think you're right that the services package is where it's at. Should we just pull the WIP mock service out of this PR, so it's not cluttering up the joint? It's not being used yet (except in the skipped test).

**Comment by travis** [comment]
>  Should we just pull the WIP mock service out of this PR

Yep that sounds great - probably time to create a new package (`@w3ui/services`?) where that can live. If you're up for creating it go for it, otherwise I'm happy to get that into a draft PR in the next day or so.

**Comment by travis** [comment]
oh also maybe just singular `@w3ui/service` since @gobengo unified the two services into one in https://github.com/web3-storage/w3protocol/issues/325#issue-1519480140

**Comment by yusefnapora** [comment]
sweet, this is passing on CI now that I bumped it up to node 18.

I'll go ahead and merge now that the mocks are gone. `@w3ui/service` sounds good to me for the new package. I can start on that tomorrow unless you feel like grabbing it sooner :)

**Pull request merged**